src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,128 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_ARRAY_SIZE = 10;
+    private int size;
+    private T[] listArray;
+
+    public ArrayList() {
+        size = 0;
+        listArray = (T[]) new Object[DEFAULT_ARRAY_SIZE];
+    }
+
     @Override
     public void add(T value) {
+        if (size >= listArray.length) {
+            listArray = extendArray();
+        }
 
+        listArray[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size + 1 || index < 0) {
```

Comment:
The condition to check the validity of the index should be 'index > size' instead of 'index > size + 1' because adding an element at index equal to the size is a valid operation in ArrayList, it's similar to append at the end.